thrust/thrust/detail/type_traits.h

```diff
@@ -60,12 +60,15 @@ namespace detail
      constexpr __host__ __device__ value_type operator()() const noexcept { return value; }
    };
 
- /// typedef for true_type
+ /// typedef for true_type, delete when no longer needed?
  typedef integral_constant<bool, true>  true_type;
 
- /// typedef for true_type
+ /// typedef for true_type, delete when no longer needed?
  typedef integral_constant<bool, false> false_type;
 
+ using true_types = cuda::std::true_type; // avoid friction with rest of codebase
```

Comment:
The idea here is that in respect of keeping PRs small, adding a temporary secondary alias would allow for this change iwthout having to modify a lot more of the codebase. The tests for `allocator.cu` will fail, for example, because libcudacxx `is_same` will not return `thrust::detail::true_type` or `thrust::detail::false_type` but `cuda::std::true_type` etc.

Reply:
If that is so then I would suggest to first start with replacing `integral_constant` and `{true, false}_type` and then follow up with `is_same`